packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -127,6 +123,24 @@ export default async function processDeployment({
           total: missingSize,
           clear: true,
         });
+
+        bar.tick(0);
+
+        uploads.forEach((e: any) =>
+          e.on('progress', () => {
+            if (!bar) return;
+
+            const totalBytesUploaded = uploads.reduce((acc: number, e: any) => {
+              return acc + e.bytesUploaded;
+            }, 0);
+            bar.curr = totalBytesUploaded;
+            bar.tick(0);
```

Comment:
**suggestion (non-blocking):** make the progress bar updating clearer

Being unfamiliar with the `progress` package, I had to dig a bit to figure out what exactly `bar.tick(0)` was doing here. Turns out, it's [rendering without advancing the counter, checking if the progress is complete](https://github.com/visionmedia/node-progress/blob/master/lib/node-progress.js#L75-L107).

---

We could add a comment about what `bar` is doing here:

```suggestion
            // set the current progress bar value
            bar.curr = totalBytesUploaded;
            // trigger rendering
            bar.tick(0);
```

---

Or, we could leverage `bar.render` directly, which wouldn't need a comment, as much:
 
```suggestion
            bar.curr = totalBytesUploaded;
            bar.render();
```

but it looks like that doesn't call the "are we done" logic. So we'd also have to add a `bar.tick(0)` before checking `bar.complete` to make it work. Or we'd have to do our own checking that `totalBytesUploaded >= missingSize`. Bah.

---

Or we could leverage `bar.update`:

```suggestion
            bar.update(totalBytesUploaded / missingSize);
```

---

Or we could leverage `bar.tick`, but using the diff:

```suggestion
            bar.tick(missingSize - totalBytesUploaded);
```

---

I got carried away with the options. Ultimately, I think this could be a bit clearer, but it's not necessary to change.

Reply:
The reason that I'm updating `curr` directly is in case there's ever a retry on a file upload, in which case we should reset the bytes uploaded count for that file - effectively "subtracting" from the progress. I think adding some comments to the existing code is fine enough.

This `progress` module is kinda "meh" though and I don't think it's really necessary. I'd like to just stick to the Ora spinner and calculate the progress manually. Wanted to do that in a follow-up PR though since this one is already busy enough.